app/controllers/actions/AuthenticateActionProvider.scala

```diff
@@ -22,11 +22,15 @@ import play.api.mvc.ActionBuilder
 import play.api.mvc.AnyContent
 import play.api.mvc.DefaultActionBuilder
 
+/*
+  This is made available for testing purposes. Do not extend this,
+  use the other actions that provide the behaviour that is needed.
+ */
 trait AuthenticateActionProvider {
   def apply(): ActionBuilder[AuthenticatedRequest, AnyContent]
 }
 
-class AuthenticateActionProviderImpl @Inject()(
+private[actions] class AuthenticateActionProviderImpl @Inject()(
   authenticate: AuthenticateAction,
   buildDefault: DefaultActionBuilder
 ) extends AuthenticateActionProvider {
```

Comment:
Could this be moved into Spec or a testOnly folder if only made for testing?

Reply:
Ah sorry, the trait is only made available so that we can create a fake, but the `AuthenticateActionProviderImpl` is what should be used